wa/workloads/sysbench/__init__.py

```diff
@@ -0,0 +1,215 @@
+                  ``max_requests`` is specified, this will default to 30
+                  seconds.
+                  '''),
+        Parameter('file_test_mode', default=None,
+                  allowed_values=['seqwr', 'seqrewr', 'seqrd', 'rndrd', 'rndwr', 'rndrw'],
+                  description='''
+                  File test mode to use. This should only be specified if
+                  ``test`` is ``"fileio"``; if that is the case and
+                  ``file_test_mode`` is not specified, it will default to
+                  ``"seqwr"`` (please see sysbench documentation for
+                  explanation of various modes).
+                  '''),
+        Parameter('cmd_params', kind=str, default='',
+                  description='''
+                  Additional parameters to be passed to sysbench as a single
+                  stiring.
+                  '''),
+        Parameter('taskset_mask', kind=int, default=0,
+                  description='''
+                  The processes spawned by sysbench will be pinned to cores as
+                  specified by this parameter.
+                  '''),
+    ]
+
+    def validate(self):
+        if (self.max_requests is None) and (self.max_time is None):
+            self.max_time = 30
+        if self.max_time and (self.max_time + 10) > self.timeout:
+            self.timeout = self.max_time + 10
+        if self.test == 'fileio' and not self.file_test_mode:
+            self.logger.debug('Test is "fileio" and no file_test_mode specified -- using default.')
+            self.file_test_mode = 'seqwr'
+        elif self.test != 'fileio' and self.file_test_mode:
+            raise ConfigError('file_test_mode must not be specified unless test is "fileio"')
+
+    @once
+    def initialize(self, context):
+        exe = Executable(self, self.target.abi, 'sysbench')
+        host_binary = context.resolver.get(exe, strict=False)
+        Sysbench.target_binary = self.target.install(host_binary)
+
+    def setup(self, context):
+        self.host_results_file = None
+        params = dict(test=self.test,
+                      num_threads=self.num_threads)
+        if self.max_requests:
+            params['max_requests'] = self.max_requests
+        if self.max_time:
+            params['max_time'] = self.max_time
+        self.target_results_file = self.target.get_workpath('sysbench_result.txt')
+        self.command = self._build_command(**params)
+
+    def run(self, context):
+        self.target.execute(self.command, timeout=self.timeout)
+
+    def extract_results(self, context):
+        self.host_results_file = os.path.join(context.output_directory, 'sysbench_result.txt')
+        self.target.pull(self.target_results_file, self.host_results_file)
+        context.add_artifact('sysbench_output', self.host_results_file, kind='raw')
+
+    def update_output(self, context):
+        if not self.host_results_file:
```

Comment:
Should this not be `if not os.exists(self.host_results_file)`?

Reply:
Yes. Yes it should.